File: C/CWE-14/2/bad.c
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */
#include "stdio.h"
#include "stdlib.h"

int main(int argc, char *argv[])
{
	int a = 1 + 2; //suppose that a is a secure variable
	printf("a = %d\n", a);
	a = 0; //dead store occurs here
	return EXIT_SUCCESS;
}

/* Corrected Function
int main(int argc, char *argv[])
{
	volatile int a = 1 + 2; //suppose that a is a secure variable
	printf("a = %d\n", a);
	a = 0; //dead store occurs here
	return EXIT_SUCCESS;
}
*/

File: C#/CWE-14/2/bad.cs
/*
 * Copyright © 2018, Jon Hood www.hoodsecurity.com
 * License: ISC
 */

using System;

namespace BadWeakness
{
	class Bad
	{
		static void Main()
		{
			string s = "MyPass";
			if (s.Equals("MyPass"))
				Console.WriteLine("User Accepted!");
			s = string.Empty; //dead store
		}
	}
}
